src/Magnum/Platform/EmscriptenApplication.cpp

```diff
@@ -506,8 +506,9 @@ void EmscriptenApplication::setupCallbacks(bool resizable) {
     emscripten_set_keydown_callback(keyboardListeningElement, this, false,
         ([](int, const EmscriptenKeyboardEvent* event, void* userData) -> Int {
             EmscriptenApplication& app = *static_cast<EmscriptenApplication*>(userData);
-            if(app.isTextInputActive() && std::strlen(event->key) == 1) {
-                TextInputEvent e{*event, {event->key, 1}};
+            const std::size_t keyLen = std::strlen(event->key);
+            if(app.isTextInputActive() && keyLen != 0) {
+                TextInputEvent e{*event, {event->key, keyLen}};
```

Comment:
Right, in my patch I forgot to change the `1` and so the UTF-8 char got sliced, producing crap :sweat_smile: 

Wait ... but won't this add stuff like Shift or F1 to text input as well?

Reply:
No, that's only in `event.code`, not `event.key`... I think, let me check

Reply:
Oh no! You're right :(

Reply:
Use my patch then :)